python/pyproject.toml

```diff
@@ -43,25 +43,24 @@ include = [{ path = "dev", format = "sdist" }]
 
 [tool.poetry.dependencies]
 python = "^3.8"
-mmhash3 = "3.0.1"
-requests = ">=2.28.1,<=2.28.2"
-click = "8.1.3"
-rich = ">=13.0.0,<=13.3.4"
-pyyaml = ">=5.4.0,<=6.0.0"
-pydantic = "1.10.7"
-fsspec = ">=2022.8.2,<=2023.3.0"
-pyparsing = "3.0.9"
-zstandard = "0.21.0"
-pyarrow = { version = ">=8.0.0,<=11.0.0", optional = true }
-pandas = { version = ">=1.4.4,<=1.5.3", optional = true }
-duckdb = { version = ">=0.6.0,<=0.7.1", optional = true }
-ray = { version = ">=2.0.0,<=2.3.1", optional = true }
-python-snappy = { version = "0.6.1", optional = true }
-thrift = { version = "0.16.0", optional = true }
-boto3 = { version = "1.24.59", optional = true }
-# The versions of the fsspec implementations should run in sync with fsspec above
-s3fs = { version = ">=2022.8.2,<=2023.3.0", optional = true }
-adlfs = { version = ">=2022.8.2,<=2023.3.0", optional = true }
+mmhash3 = ">=3.0.0,<4.0.0"
+requests = ">=2.20.0,<3.0.0"
+click = ">=7.1.1,<9.0.0"
+rich = ">=10.11.0,<14.0.0"
+pyyaml = ">=5.4.0,<6.0.0" # CVE-2020-14343 was fixed in 5.4.
```

Comment:
Should we clean up these justifications as well?

Reply:
I left this one as a reasonable answer as to why the 5.4.0 lower bound shouldn't be reduced (should somebody want to reevaluate the bounds later), but I'm happy to remove this comment if preferred. I cleaned up the more nebulous justifications. Your call on this.